flycheck.el

```diff
@@ -9067,6 +9067,18 @@ Requires Sphinx 1.2 or newer.  See URL `http://sphinx-doc.org'."
   :predicate (lambda () (and (flycheck-buffer-saved-p)
                              (flycheck-locate-sphinx-source-directory))))
 
+(defun flycheck-ruby--find-project-root (_checker)
+  "Come up with a suitable default directory to run CHECKER in.
+
+This will either be the directory that contains `Gemfile' or,
+if no such file is found in the directory hierarchy, Flycheck will
+use the value of `default-directory'."
+  (or
+   (and
+    buffer-file-name
+    (locate-dominating-file buffer-file-name "Gemfile"))
+   default-directory))
```

Comment:
Do you need the `(or … default-directory)` part? I think `:working-directory` functions can return nil.
I'd also update the docstring to not mention `CHECKER`, since that's not what the function does. I'd write something like this:

> Compute an appropriate working-directory for flycheck-ruby.
>
> This is either a parent directory containing a Gemfile, or nil.

Reply:
@cpitclaudel done. Also fixed one test. Can you review this again?